### DISCUSSION TITLE
MySQL connect ECONNREFUSED 127.0.0.1:3306

### DISCUSSION THREAD

**Discussion by Al-Blu** [Budibase/budibase#5211] [Help]
Hi All, 
I am struggling to create a MySQL DataSource on a MacBook. I get the following error
![Screenshot 2022-04-01 at 14 35 41](https://user-images.githubusercontent.com/99728863/161264832-ce39ba9b-bab1-4b2b-8819-02303e331ab1.png)

I went back on a few other discussions here but cannot find a solution. 
I am not hosting with Docker. 
I can ping localhost
I am hosting at "https://MyCompany.budibase.app/builder/app/XXXX/data/datasource/bb_internal"
I have turned off the firewall and tried another MacBook

cat /etc/hosts results in 
           127.0.0.1	localhost
           255.255.255.255	broadcasthost
           ::1             localhost

I can connect to MySQL with MySQL Workbench and the connection is localhost & 3306 
![Screenshot 2022-04-01 at 14 40 54](https://user-images.githubusercontent.com/99728863/161266471-90ae1426-565e-4d9e-a40d-040a3f6cedc5.png)

Any other ideas as to why I cannot access MySQL ??
Thanks

**Comment by melohagan** [comment]
Hi @Al-Blu 

Are you hosting budibase inside docker? In that case you will need to set the host to: host.docker.internal

**Reply by Mike-Carolan** [reply]
I had this same exact thing with (except with Postgres).  Changing the host to host.docker.internal fixed it.  This happens because Docker is actually driving the application layer - not Budibase.  Once you deploy the application (outside of Docker), then you can change the host back to localhost and it will work just fine.

**Reply by MoallaMed** [reply]
Hello
 You changed it in the docker-compose.yaml or where exactly  ?
Thanks

**Reply by melohagan** [reply]
Hey @MoallaMed If you look at the screenshot at the top of the discussion, you'd be changing the `Host: localhost` to `Host: host.docker.internal`

**Comment by Al-Blu** [comment]
I am not hosting it 0n Docker. I am using the BudiBase hosting for now as we are only in development.
"**https://mycompany.budibase.app/builder/app/XXXX/data/datasource/bb_internal**"
But if I did would it on Docker would that be a solution?

**Reply by melohagan** [reply]
Ah I see. If you are hosting budibase remotely, then you wont be able to connect to the localhost of your machine.
 
If you hosted budibase locally in docker, then you will need to use host.docker.internal 
Otherwise, you will need to deploy your MySQL instance with a publicly available host connection.
